hackathon_site/dashboard/frontend/src/pages/Inventory/Inventory.test.tsx

```diff
@@ -34,13 +35,48 @@ describe("Inventory Page", () => {
             .calledWith(categoriesUri)
             .mockResolvedValue(categoryApiResponse);
 
-        const { getByText } = render(<Inventory />);
+        const { getByText, queryByText } = render(<Inventory />);
 
         await waitFor(() => {
             expect(getByText(mockHardware[0].name)).toBeInTheDocument();
             expect(getByText(mockCategories[0].name)).toBeInTheDocument();
         });
 
+        // by default the hardwareUri has a limit of 100
+        if (mockHardware.length <= 100) {
+            expect(queryByText(/load more/i)).not.toBeInTheDocument();
+        } else {
+            expect(getByText(/load more/i)).toBeInTheDocument();
+        }
+    });
+
+    test("Hardware count", async () => {
+        // Mock hardware api response with a limit of mockHardware.length - 2
+        const limit = mockHardware.length - 2;
+        const hardwareApiResponse = makeMockApiListResponse(
+            mockHardware.slice(0, limit),
+            null,
+            null,
+            mockHardware.length
+        );
+
+        when(mockedGet)
+            .calledWith(hardwareUri, {})
+            .mockResolvedValue(hardwareApiResponse);
+
+        const { getByText, getByTestId } = render(<Inventory />);
+        const { getByText: getTotalCountText } = within(
+            getByTestId("inventory-total-count")
```

Comment:
When it comes to UI testing, the general guideline is "test what users see". `getByTestId` breaks that principle, and should really only be used as a last resort when things like `getByText` can't be used (such as for testing that loading bars appear).

In this case, do we absolutely need to use `getByTestId`? Or would it be sufficent just to use the regular `getByText()` on the entire page? In all of our other tests, we don't really test _where_ things appear, only _that_ they appear. Too much of the former leads back to snapshot tests, which get annoying when you want to update the UI or something that shouldn't break functional tests like these.

Reply:
I see, I see (pun intended), I will use getByText().